tools/dfx-orbit/src/dfx_extension_api.rs

```diff
@@ -0,0 +1,126 @@
+        Self::init_extensions_dir(user_config_dir)
+    }
+
+    fn init_extensions_dir(path: PathBuf) -> anyhow::Result<cap_std::fs::Dir> {
+        let extensions_dir = path.join("extensions");
+        std::fs::create_dir_all(&extensions_dir).with_context(|| {
+            format!(
+                "Could not create directory at: {}",
+                extensions_dir.display()
+            )
+        })?;
+        let std_dir = std::fs::File::open(&extensions_dir).with_context(|| {
+            format!("Could not open directory at: {}", extensions_dir.display())
+        })?;
+        let cap_dir = cap_std::fs::Dir::from_std_file(std_dir);
+        Ok(cap_dir)
+    }
+
+    /// Gets the basename of the extension config file.
+    fn config_file_name(&self) -> String {
+        format!("{}.json", ORBIT_EXTENSION_NAME)
+    }
+
+    /// Gets the extension config file for this extension.  If the file does not exist, it will be created.
+    ///
+    /// E.g. `~/.config/dfx/extensions/<extension_name>.json`
+    ///
+    /// Note: The file SHOULD be JSON but this is not enforced.
+    pub(crate) fn extension_config_file(&self) -> anyhow::Result<cap_std::fs::File> {
+        let extension_config_dir = &self.extensions_dir;
+        let filename = self.config_file_name();
+        let mut open_options = cap_std::fs::OpenOptions::new();
+        let open_options = open_options.read(true).write(true).create(true);
+        extension_config_dir
+            .open_with(filename, open_options)
+            .with_context(|| {
+                format!(
+                    "Could not create extension config file for extension: {}",
+                    ORBIT_EXTENSION_NAME
+                )
+            })
+    }
+
+    /// Gets the extension config directory for this extension.
+    pub(crate) fn extension_config_dir(&self) -> anyhow::Result<cap_std::fs::Dir> {
+        let extensions_dir = &self.extensions_dir;
+        extensions_dir
+            .create_dir_all(ORBIT_EXTENSION_NAME)
+            .with_context(|| {
+                format!(
+                    "Could not create extension directory for extension: {}",
+                    ORBIT_EXTENSION_NAME
+                )
+            })?;
+        extensions_dir
+            .open_dir(ORBIT_EXTENSION_NAME)
+            .with_context(|| {
+                format!(
+                    "Could not open extension directory for extension: {}",
+                    ORBIT_EXTENSION_NAME
+                )
+            })
+    }
+
+    /// Gets the dfx_core interface
+    pub async fn dfx_interface(&mut self) -> anyhow::Result<DfxInterface> {
+        let network_name = self
+            .station_or_default(None)
+            .with_context(|| "Failed to get station")?
+            .network;
+        let interface_builder = DfxInterface::builder().with_network_named(&network_name);
+        let interface = interface_builder.build().await?;
+        if !interface.network_descriptor().is_ic {
```

Comment:
do you know how `is_ic` is determined? would this work if the `dfx.json` has a custom section of networks that also point at mainnet? e.g. staging, testing, production environments

Reply:
If I understand correctly, it should work, as long as the proveder points to "icp0.io":
https://github.com/dfinity/sdk/blob/master/src/dfx-core/src/config/model/network_descriptor.rs#L80

Reply:
I see, ideally that dfx code would also check for `icp-api.io`, but for now i think it's fine then

Reply:
Well, there are some upstream changes that I need to propose to the `sdk` team anyway. I can bring that up  too